### PR TITLE
Restore tests

### PR DESCRIPTION
**Pull request by smalls**
This continues the work that @shaper started in #709.

I added the shell tests to Travis - this is a bit of an experiment, and there's a failing test on my machine (that doesn't seem related to the refactor), so we'll see how it goes.

Still, it's looking like the tests are working:

screenshot1

And! As a bonus, zooming out shows:

screenshot zoom out

### CODE REVIEW

**Comment by smalls** [comment]
This is a good chunk of #711.

**Comment by smalls** [comment]
Current state of things.. `npm test` now starts the server & runs gulp. Especially the latter is a bit slow, so I've added support for `npm run test-test` that won't start the server or run gulp.

I've filed https://github.com/PolymerLabs/arcs/issues/729 to revisit the gulp issue. We'll probably want to keep starting the server though.

**Comment by smalls** [comment]
There's a test failure but it seems legit so I'm going to merge (as long as my recent changes still look good to @shaper).

I'll file a bug for the test failure.

File: shell/package.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{
+  "name": "shell",
+  "version": "0.0.0",
+  "private": true,
+  "config": {
+    "port": "8080",
+    "host": "localhost"
+  },
+  "scripts": {
+    "start": "cross-env http-server -a ${npm_package_config_host} -p ${npm_package_config_port}",
```
Comment (shaper) [inline shell/package.json]: Add comment explaining that we're sort of using npm as a build system here, and we may want to rework this logic to fit within either sigh or gulp.

https://stackoverflow.com/questions/14221579/how-do-i-add-comments-to-package-json-for-npm-install

**Comment by smalls** [comment]
I filed #731 for the failure.

**Pull request merged**